cmd/roachtest: add first version roachtest for tenant streaming

gh-casper opened the pull request:
This initial version of roachtest creates two crdb clusters,
and sets up stream replication between the two, runs tpcc or
ledge workload on the source, and verify if two tenants are
identical after the stream replication.

This PR also adds InitTarget as roachprod cluster option to
support multiple crdb clusters in roachprod.

Release note: None

gh-casper committed:
roachprod: support InitTarget when starting nodes in roachprod cluster

Release note (cli change): add init-target flag in roachprod start
and change roachprod init to init per node.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80846)
<!-- Reviewable:end -->

gh-casper committed:
cmd/roachtest: add first version roachtest for tenant streaming

This initial version of roachtest creates two crdb clusters,
and sets up stream replication between the two, runs tpcc or
ledge workload on the source, and verify if two tenants are
identical after the stream replication.

Release note: None

shermanCRL commented:
Is this PR still relevant, or handled by other PRs?
